mesonbuild/modules/external_project.py

```diff
@@ -93,10 +93,10 @@ def __init__(self,
         # will install files into "c:/bar/c:/foo" which is an invalid path.
         # Work around that issue by removing the drive from prefix.
         if self.prefix.drive:
-            self.prefix = self.prefix.relative_to(self.prefix.drive)
+            self.prefix = Path(relpath(self.prefix, self.prefix.drive))
 
         # self.prefix is an absolute path, so we cannot append it to another path.
-        self.rel_prefix = self.prefix.relative_to(self.prefix.root)
+        self.rel_prefix = Path(relpath(self.prefix, self.prefix.root))
```

Comment:
Using `relative_to(self.prefix.anchor)` should also work in both cases as far as I understand. https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.anchor